pkg/enclave_encrypt/server.go

```diff
@@ -52,40 +59,71 @@ func (s *EnclaveEncryptServer) Encrypt(clientTarget []byte, plaintext []byte) (*
 	if err != nil {
 		return nil, err
 	}
+	data := ServerSendData{
+		EncappedPublic: Bytes(encappedPublic),
+		Ciphertext:     Bytes(ciphertext),
+		OrganizationId: s.organizationId,
+		UserId:         s.userId,
+	}
 
-	encappedPublicSignature, err := P256Sign(s.enclaveAuthKey, encappedPublic)
+	dataBytes, err := json.Marshal(data)
 	if err != nil {
 		return nil, err
 	}
+	dataSignature, err := P256Sign(s.enclaveAuthKey, dataBytes)
+	if err != nil {
+		return nil, err
+	}
+	dataSig := Bytes(dataSignature)
 
-	encSig := Bytes(encappedPublicSignature)
-	enc := Bytes(encappedPublic)
-	ciph := Bytes(ciphertext)
-	return &ServerSendMsg{
-		EncappedPublic:          &enc,
-		EncappedPublicSignature: &encSig,
-		Ciphertext:              &ciph,
+	enclaveQuorumPublic := s.enclaveAuthKey.PublicKey
+	// FIXME: `elliptic.Marshal` is deprecated,
```

Comment:
any context on this? are there security reasons why we shouldn't be using it?

Reply:
from what i've read, it's deprecated in favor of crypto/ecdh (for ECDH) due to lower performance during encryption. since we use `elliptic.Marshal` in a few other places, i think we can do a separate refactor/testing of all those instances that we use `elliptic.Marshal`/`elliptic.Unmarshal` as a follow up.